pkg/users/rename-group-dialog.jsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React from 'react';
+import { Form, FormGroup } from "@patternfly/react-core/dist/esm/components/Form/index.js";
+import { HelperText, HelperTextItem } from "@patternfly/react-core/dist/esm/components/HelperText/index.js";
+import { TextInput } from "@patternfly/react-core/dist/esm/components/TextInput/index.js";
+
+import { apply_modal_dialog, show_modal_dialog } from "cockpit-components-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+function RenameGroupDialogBody({ state, change }) {
+    const { name } = state;
+
+    return (
+        <Form isHorizontal onSubmit={apply_modal_dialog}>
+            <FormGroup fieldId="group-name" label={_("New name")}>
+                <TextInput id="group-name" onChange={val => change("name", val)} value={name} />
+            </FormGroup>
+            <HelperText>
+                <HelperTextItem variant="warning">{_("There might be permission-based side effects associated with group renaming")}</HelperTextItem>
```

Comment:
I'm not super happy about the text. This is primarily about sudo and polkit, right? Maybe

> Renaming a group may affect sudo and similar rules

Also, I keep forgetting -- period or not? @garrett -- any opinion about making this shorter, more useful, and easier to read and translate?